packages/workspace/src/workspace/state/static_files_sources/history_static_files_source.ts

```diff
@@ -41,23 +41,26 @@ export const buildHistoryStateStaticFilesSource = (
 
   return {
     persistStaticFile: async (file: StaticFile): Promise<void> => {
+      const path = getFileName(file.filepath, file.hash)
+      const existingFiles = await listFiles()
+      if (existingFiles.has(path)) {
+        return
+      }
+
       const content = await file.getContent()
       if (content === undefined) {
         log.warn(`Received file ${file.filepath} to set without content`)
         return
       }
-      const path = getFileName(file.filepath, file.hash)
-      if (!(await listFiles()).has(path)) {
-        await dirStore.set({
-          buffer: content,
-          filename: getFileName(file.filepath, file.hash),
-        });
-        (await listedFilesCache)?.add(getFileName(file.filepath, file.hash))
-      }
+      await dirStore.set({
+        buffer: content,
+        filename: getFileName(file.filepath, file.hash),
+      })
+      existingFiles.add(getFileName(file.filepath, file.hash))
     },
     getStaticFile: async (path, encoding, hash) => {
       if (hash === undefined) {
-        throw new Error('Hash must be passed to getStaticFile of HistoryStaticFilesSource')
+        throw new Error(`path ${path} was passed without an hash to getStaticFile of HistoryStaticFilesSource`)
```

Comment:
english nit
```suggestion
        throw new Error(`path ${path} was passed without a hash to getStaticFile of HistoryStaticFilesSource`)
```
also, the file name is always part of the log (the logger was initialized with `const log = logger(module)`), there is no need to specify the `HistoryStaticFilesSource` part in the log